splice_si_api/src/main/java/com/splicemachine/si/impl/txn/WritableTxn.java

```diff
@@ -129,46 +131,66 @@ public State getState(){
     public void commit() throws IOException{
         if(LOG.isTraceEnabled())
             SpliceLogUtils.trace(LOG,"Before commit: txn=%s",this);
-        switch(state){
-            case COMMITTED:
-                return;
-            case ROLLEDBACK:
-                throw exceptionFactory.cannotCommit(txnId,state);
-        }
-        synchronized(this){
-            //double-checked locking for efficiency--usually not needed
-            switch(state){
+
+        if (getSubId() == 0) {
+            switch (state) {
                 case COMMITTED:
                     return;
                 case ROLLEDBACK:
-                    throw exceptionFactory.cannotCommit(txnId,state);
+                    throw exceptionFactory.cannotCommit(txnId, state);
+            }
+            synchronized (this) {
+                //double-checked locking for efficiency--usually not needed
+                switch (state) {
+                    case COMMITTED:
+                        return;
+                    case ROLLEDBACK:
+                        throw exceptionFactory.cannotCommit(txnId, state);
```

Comment:
Nice to add default to switch stmt or convert to if-else if there are only 2 states